tests/unit/api/test_template.py

```diff
@@ -94,3 +98,73 @@ def test_evaluate(client, building_motif):
     assert (BLDG["zone1"], A, BRICK.HVAC_Zone) in graph
     assert (BLDG["zone1"], BRICK.isFedBy, BLDG["cav1"]) in graph
     assert len(list(graph.triples((None, None, None)))) == 3
+
+
+def test_evaluate_bad_templated_id(client, building_motif):
+    model = Model.create(name="urn:my_model")
+
+    results = client.post(
+        "/templates/-1/evaluate",
+        json={
+            "model_id": model.id,
+            "bindings": {"name": {"@id": BLDG["zone1"]}, "cav": {"@id": BLDG["cav1"]}},
+        },
+    )
+
+    assert results.status_code == 404
+
+
+def test_evaluate_no_body(client, building_motif):
+    lib = Library.load(directory="tests/unit/fixtures/templates")
+    zone = lib.get_template_by_name("zone")
+    zone.inline_dependencies()
+    assert zone.parameters == {"name", "cav"}
+
+    results = client.post(f"/templates/{zone.id}/evaluate")
+
+    assert results.status_code == 400
+
+
+def test_evaluate_bad_body(client, building_motif):
+    model = Model.create(name="urn:my_model")
+    lib = Library.load(directory="tests/unit/fixtures/templates")
+    zone = lib.get_template_by_name("zone")
+    zone.inline_dependencies()
+    assert zone.parameters == {"name", "cav"}
+
+    results = client.post(
+        f"/templates/{zone.id}/evaluate",
+        json={
+            # no model
+            "bindings": {"name": {"@id": BLDG["zone1"]}, "cav": {"@id": BLDG["cav1"]}},
+        },
+    )
+
+    assert results.status_code == 400
+
+    results = client.post(
+        f"/templates/{zone.id}/evaluate",
+        json={
+            "model_id": model.id,
+            # no bindings
+        },
+    )
+
+    assert results.status_code == 400
+
+
+def test_evaluate_bad_model_id(client, building_motif):
+    lib = Library.load(directory="tests/unit/fixtures/templates")
+    zone = lib.get_template_by_name("zone")
+    zone.inline_dependencies()
+    assert zone.parameters == {"name", "cav"}
+
+    results = client.post(
+        f"/templates/{zone.id}/evaluate",
+        json={
+            "model_id": -1,
+            "bindings": {"name": {"@id": BLDG["zone1"]}, "cav": {"@id": BLDG["cav1"]}},
+        },
+    )
+
+    assert results.status_code == 404
```

Comment:
I think there could be one more test for no `model_id`

Reply:
that's tested in `test_evaluate_bad_body`, L142